app/controllers/download_original_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class DownloadOriginalController < ApplicationController
+  def stage
+    request = params
+    begin
+      child_object = ChildObject.find(request['oid'].to_i)
+    rescue ActiveRecord::RecordNotFound
+      render(json: { "title": "Invalid Child OID" }, status: 400) && (return false)
+    end
+    return unless check_child_visibility(child_object)
+    SaveOriginalToS3Job.perform_later(child_object.oid)
+    render(json: { "title": "Child object staged for download." }, status: 200)
+  end
+
+  def check_child_visibility(child_object)
+    render(json: { "title": "Child Object is restricted." }, status: 403) && (return false) if child_object.parent_object.visibility == "Private" || child_object.parent_object.visibility == "Redirect"
```

Comment:
could we change this to check for statuses that are allowed, so it 'fails safe' if we introduce a new one?